### ISSUE TITLE
how to split train and test dataset

### ISSUE THREAD

**Issue by ZYJ-JMF** [ecnuycxie/DG-Font#16]
Hi,I am wondering how to split the train and test datasets?
In your paper,it said that "The dataset is randomly partitioned into a training set and testing set".
Dose randomly mean the Font content test set in differents Font style is different?

**Comment by ecnuycxie** [comment]
Thank you for your question. In order to ensure that the characters in the test set are invisible during the training process, the content in the test set of each style is the same.

**Comment by ZYJ-JMF** [comment]
Thanks for your answer.
I have another question，how to evaluation the performance of the result and get the quantitative score?
In the validataion, every font(total 8) can generate a series of styled fonts.Which is used for the eval? or you caculate the average score?
Thanks

**Comment by ecnuycxie** [comment]
I used the font KAI as the content during the testing process. You can also use other fonts as content, such as SONG

**Comment by ZYJ-JMF** [comment]
oh,Thanks for your answer.
I want to make sure whether I understand your answer.
DO you mean that you use other style font as the content to generate the KAI style and compare with the groundtruth font KAI?

**Comment by ecnuycxie** [comment]
In the test process, we use KAI as the source font (content) to generate other styles and compare these generated images with their ground-truth.

**Comment by ZYJ-JMF** [comment]
"In the test process, we use KAI as the source font (content) to generate other styles and compare these generated images with their ground-truth."
" other styles" here are choosed from the total 990 word or from the valid 190 words as the reference style?
Do you random choose 50 words as the reference and  generated images?
Will you share the eval code?
Thanks

**Comment by dagongji10** [comment]
@ecnuycxie 
I also have some questions about the split of dataset.
- The Paper said: training set contains 400 fonts and each font contains 800 characters, testing set contains remaining 190 characters of the 400 fonts and remaining 10 fonts.
- But in the [code](https://github.com/ecnuycxie/DG-Font/blob/099e3c9aafb6c172afa1a4f7c6c7e80c495b7b33/datasets/datasetgetter.py#L58): training set contains 410 fonts and each font contains 800 characters, testing set contains remaining 190 characters of the 410 fonts.

Is am I right? How to understand this difference? Is this make any difference for DG-Font?

**Comment by Hryxyhe** [comment]
> [@ecnuycxie](https://github.com/ecnuycxie) I also have some questions about the split of dataset.
> 
> * The Paper said: training set contains 400 fonts and each font contains 800 characters, testing set contains remaining 190 characters of the 400 fonts and remaining 10 fonts.
> * But in the [code](https://github.com/ecnuycxie/DG-Font/blob/099e3c9aafb6c172afa1a4f7c6c7e80c495b7b33/datasets/datasetgetter.py#L58): training set contains 410 fonts and each font contains 800 characters, testing set contains remaining 190 characters of the 410 fonts.
> 
> Am I right? How to understand this difference? Does that make any difference for DG-Font?
> 
> Another question, [validation](https://github.com/ecnuycxie/DG-Font/blob/099e3c9aafb6c172afa1a4f7c6c7e80c495b7b33/validation/validation.py#L94) seems only save generated images, no useful information generated for training. So can I comment out [this line](https://github.com/ecnuycxie/DG-Font/blob/099e3c9aafb6c172afa1a4f7c6c7e80c495b7b33/main.py#L250) for speeding up training？

Yes , I found the same difference. The code did not make test font unseen during train, but only make the characters unseen
